ocaml/xapi/xapi_pool_update.ml

```diff
@@ -340,20 +340,39 @@ let introduce ~__context ~vdi =
 let pool_apply ~__context ~self =
   let pool_update_name = Db.Pool_update.get_name_label ~__context ~self in
   debug "pool_update.pool_apply %s" pool_update_name;
-  ()
+  Db.Pool_update.get_hosts ~__context ~self |>
+  List.set_difference (Db.Host.get_all ~__context) |>
+  List.iter (fun host ->
+    ignore(Helpers.call_api_functions ~__context
+      (fun rpc session_id -> Client.Pool_update.apply rpc session_id self host))
+  )
 
 let clean ~__context ~self ~host =
   let pool_update_name = Db.Pool_update.get_name_label ~__context ~self in
   let host_name = Db.Host.get_name_label ~__context ~self:host in
   debug "pool_update.clean %s on %s" pool_update_name host_name;
-  ()
+  if Pool_role.is_master () then begin
```

Comment:
Hmm, are we going do need to do any more than this in this function? If not, we should drop the `host` parameter